Gir_GstGL.toml

```diff
@@ -0,0 +1,225 @@
+[options]
+girs_dir = "gir-files"
+library = "GstGL"
+version = "1.0"
+min_cfg_version = "1.0"
+target_path = "gstreamer-gl"
+work_mode = "normal"
+concurrency = "send+sync"
+generate_safety_asserts = true
+single_version_file = true
+
+external_libraries = [
+    "GLib",
+    "GObject",
+    "Gst",
+    "GstBase",
+    "GstVideo",
+]
+
+generate = [
+    # BitFlags
+    "GstGL.GLAPI",
+    "GstGL.GLDisplayType",
+    "GstGL.GLPlatform",
+    #"GstGL.GLSLProfile", # No Shader Language for now
+    # Enums
+    "GstGL.GLFormat",
+    "GstGL.GLQueryType",
+    #"GstGL.GLSLVersion", # FIXME: invalid symbol autogenerated
+    #"GstGL.GLTextureTarget", # FIXME: invalid symbol autogenerated
+    "GstGL.GLStereoDownmix",
+    "GstGL.GLUploadReturn",
+    # Enums Errors
+    #"GstGL.GLBaseMemoryError", # cannot find function gst_gl_base_buffer_error_quark()
+    "GstGL.GLContextError",
+    #"GstGL.GLSLError", # No Shader Language for now
+    "GstGL.GLWindowError",
+    # Records
+    #"GstGL.GLAllocationParams",
+    #"GstGL.GLRenderbufferAllocationParams",
+    #"GstGL.GLVideoAllocationParams",
+    # Objects
+    "GstGL.GLBaseFilter",
+    #"GstGL.GLBaseMemoryAllocator",
+    #"GstGL.GLBufferPool",
+    #"GstGL.GLFramebuffer",
+    #"GstGL.GLMemoryAllocator",
+    #"GstGL.GLMemoryPBOAllocator",
+    #"GstGL.GLRenderbufferAllocator",
+]
+
+manual = [
+    "GLib.Error",
+    "GLib.Thread",
+    "GObject.Object",
+    "Gst.Object",
+    "Gst.FlowReturn",
+    "Gst.PadDirection",
+    "GstVideo.VideoMultiviewFlags",
+    "GstVideo.VideoMultiviewMode",
+]
+
+[[object]]
+name = "Gst.Structure"
+status = "manual"
+ref_mode = "ref-mut"
+
+[[object]]
+name = "Gst.Buffer"
+status = "manual"
+ref_mode = "ref"
+
+[[object]]
+name = "Gst.Caps"
+status = "manual"
+ref_mode = "ref"
+
+[[object]]
+name = "Gst.Query"
+status = "manual"
+ref_mode = "ref"
+
+[[object]]
+name = "GstGL.GLBuffer"
+status = "manual"
+ref_mode = "ref"
+
+[[object]]
+name = "GstGL.GLQuery"
+status = "manual"
+ref_mode = "ref"
+
+[[object]]
+name = "GstGL.GLContext"
+status = "generate"
+trait = false
+
+    [[object.function]]
+    name = "new_wrapped"
+    # input handle
+    ignore = true
+
+    [[object.function]]
+    name = "get_gl_context"
+    # return handle
+    ignore = true
+
+    [[object.function]]
+    name = "get_current_gl_context"
+    # return handle
+    ignore = true
+
+    [[object.function]]
+    name = "get_proc_address"
+    ignore = true
+
+    [[object.function]]
+    name = "default_get_proc_address"
+    ignore = true
+
+    [[object.function]]
+    name = "get_proc_address_with_platform"
+    ignore = true
+
+    [[object.function]]
+    name = "thread_add"
+    ignore = true
```

Comment:
I want to avoid this:

```
+    //pub fn thread_add<P: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(&self, func: /*Unknown conversion*//*Unimplemented*/GLContextThreadFunc, data: P) {                                              
+    //    unsafe { TODO: call ffi::gst_gl_context_thread_add() }
+    //}
```

But, yes, perhaps we would need an unsafe call to offer a callback in rust

Reply:
It could be safe though :)